google/cloud/bigquery_storage_v1/writer.py

```diff
@@ -106,154 +106,44 @@ def __init__(
                 Extra headers to include when sending the streaming request.
         """
         self._client = client
-        self._closing = threading.Lock()
         self._closed = False
         self._close_callbacks = []
-        self._futures_queue = queue.Queue()
         self._metadata = metadata
+        self._thread_lock = threading.RLock()
+
+        self._stream_name = None
 
         # Make a deepcopy of the template and clear the proto3-only fields
         self._initial_request_template = _process_request_template(
             initial_request_template
         )
 
-        # Only one call to `send()` should attempt to open the RPC.
-        self._opening = threading.Lock()
-
-        self._rpc = None
-        self._stream_name = None
-
-        # The threads created in ``._open()``.
-        self._consumer = None
+        self._connection = _Connection(
+            client=client,
+            writer=self,
+            metadata=metadata,
+        )
 
     @property
     def is_active(self) -> bool:
```

Comment:
Is this method meant for use by the user?  Do we care that the they could see this transition from true -> false -> true during connection transitions?